### DISCUSSION TITLE
checkpoint encryption

### DISCUSSION THREAD

**Discussion by GBrics** [Xilinx/RapidWright#363] [Q&A] [answered]
Hello, 

I want to use RapidWright to analyses one of my designs that include a Xilinx IP from Vivado Catalog. 
When I export the checkpoint and try to read it with ``readCheckpoint()`` , error is raised stating that it cannot read encrypted edif files.

The Xilinx IP used is totally independent from the rest of the design and I would like to be able to export a checkpoint only for another part of the design (free from encrypted IP).

Even if I specify the corresponding hierarchical cell with the ``write_checkpoint`` tcl command, It generates an encrypted edif file as soon as the Xilinx IP is implemented somewhere else in the design.

What would be the best solution to export this part of the design as unencrypted checkpoint file ?

Thank you in advance for your help !

Gbrics

**Comment by clavin-xlnx** [comment, marked as answer]
Hi Gbrics,

Is it possible that you are using Vivado 2020.2?  There was an issue in that version that caused the entire EDIF file to be written out as encrypted if the design contained any encrypted cells.  The typical behavior is for only the encrypted cells to be written out as encrypted (`.edn` files) and the rest of the design in plaintext ASCII (`.edf`).  The issue is described here: 

https://support.xilinx.com/s/article/76191?language=en_US

The patch only can be applied if you have 2020.2.2, but any other version does not experience this issue (2021.1, 2021.2, 2020.1, etc). 

Let me know if that is not the case and we can dig deeper.

Chris

**Comment by GBrics** [comment]
Hi Chris, 

Indeed, I was using Vivado 2020.2 and my problem was exactly what you described.

I tried to apply the patch but it didn't solve the issue (not 100% sure the patch was correctly installed). Anyway, I moved to Vivado 2021.2 and now it works perfectly as intented.

Thank you very much for your help !

Gbrics
